client/scripts/lib/markdownShortcuts.js

```diff
@@ -31,220 +32,223 @@
-    const [line, offset] = this.quill.getLine(selection.index)
-    const text = line.domNode.textContent
-    const lineStart = selection.index - offset
-    if (this.isValid(text, line.domNode.tagName)) {
-      for (let match of this.matches) {
-        const matchedText = text.match(match.pattern)
-        if (matchedText) {
-          // We need to replace only matched text not the whole line
-          match.action(text, selection, match.pattern, lineStart)
-          return
-        }
-      }
-    }
-  }
-
-  onEnter () {
-    let selection = this.quill.getSelection()
-    if (!selection) return
-    if (this.options.suppress && this.options.suppress()) return
-    const [line, offset] = this.quill.getLine(selection.index)
-    const text = line.domNode.textContent + ' '
-    const lineStart = selection.index - offset
-    selection.length = selection.index++
-    if (this.isValid(text, line.domNode.tagName)) {
-      for (let match of this.matches) {
-        const matchedText = text.match(match.pattern)
-        if (matchedText) {
-          match.action(text, selection, match.pattern, lineStart)
-          return
-        }
-      }
-    }
-  }
-}
+ export class MarkdownShortcuts {
+   constructor (quill, options) {
+     this.quill = quill
+     this.options = options
+ 
+     this.ignoreTags = ['PRE']
+     this.matches = [
+       {
+         name: 'header',
+         pattern: /^(#){1,2}\s/g,
+         action: (text, selection, pattern) => {
+           var match = pattern.exec(text)
+           if (!match) return
+           const size = match[0].length
+           // Need to defer this action https://github.com/quilljs/quill/issues/1134
+           setTimeout(() => {
+             this.quill.formatLine(selection.index, 0, 'header', size - 1, 'api')
+             this.quill.deleteText(selection.index - size, size, 'api')
+           }, 0)
+         }
+       },
+       {
+         name: 'blockquote',
+         pattern: /^(>)\s/g,
+         action: (text, selection) => {
+           // Need to defer this action https://github.com/quilljs/quill/issues/1134
+           setTimeout(() => {
+             this.quill.formatLine(selection.index, 1, 'blockquote', true, 'api')
+             this.quill.deleteText(selection.index - 2, 2, 'api')
+           }, 0)
+         }
+       },
+       {
+         name: 'code-block',
+         pattern: /^`{3}/g,
+         action: (text, selection) => {
+           // Need to defer this action https://github.com/quilljs/quill/issues/1134
+           setTimeout(() => {
+             this.quill.formatLine(selection.index, 1, 'code-block', true, 'api')
+             this.quill.deleteText(selection.index - 3, 3, 'api')
+           }, 0)
+         }
+       },
+       {
+         // bolditalic now handles bold, italic, and bold italic
+         name: 'bolditalic',
+         pattern: /(?<!\S)((?:\*|_){1,3})([^\s*_]{1,2}|[^\s_*][^_*]+?[^\s_*])((?:\*|_){1,3})(?!\S)/g,
```

Comment:
For some reason the Github changes preview is failing to highlight the actual changes on this file—see line 82 pattern change, and the `for` loop starting on line 86.

Reply:
thanks for this